src/tools/ci/test_pkg_conf

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import argparse
+import shutil
+import subprocess
+import os
+import sys
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('--rootdir', help='root directory', type=str, required=True)
+parser.add_argument('--pkgconf_installdir', help='pkgconf install directory', type=str, required=True)
+parser.add_argument('--builddir', help='build directory', type=str, required=True)
+parser.add_argument('--meson', help='path to meson binary', type=str, required=True)
+
+args = parser.parse_args()
+
+rootdir = os.path.realpath(args.rootdir)
+builddir = os.path.realpath(args.builddir)
+pkgconf_installdir = os.path.realpath(args.pkgconf_installdir)
+adminpackdir = os.path.join(rootdir, 'contrib/adminpack')
+workdir = os.path.join(builddir, 'contrib/adminpack_tmp')
+exit_code = 0
+
+adminpack_meson_build_file = \
+  '''
+  project('adminpack', 'c')
+
+  pg_ext = dependency('postgresql-extension-warnings')
+
+  adminpack = shared_module('adminpack',
+    ['adminpack.c'],
+    dependencies: pg_ext,
+    name_prefix: '',
+    install_dir: pg_ext.get_variable(pkgconfig: 'dir_mod')
+  )
+
+  install_data(
+    'adminpack.control',
+    'adminpack--1.0.sql',
+    'adminpack--1.0--1.1.sql',
+    'adminpack--1.1--2.0.sql',
+    'adminpack--2.0--2.1.sql',
+    install_dir: pg_ext.get_variable(pkgconfig: 'dir_data')
+  )
+  '''
+
+# copy adminpack
+if os.path.exists(workdir):
+    shutil.rmtree(workdir)
+shutil.copytree(adminpackdir, workdir)
```

Comment:
No need to copy the tree - you can just adjust the meson.build to reference the files with '../adminpack/adminpack.c' etc.